src/storage/hummock_trace/src/collector.rs

```diff
@@ -0,0 +1,408 @@
+        let (writer_tx, writer_rx) = unbounded();
+
+        let writer_handle = tokio::spawn(GlobalCollector::start_writer_worker(writer_rx, writer));
+
+        let rx = self.rx.clone();
+
+        let collect_handle = tokio::spawn(GlobalCollector::start_collect_worker(rx, writer_tx));
+
+        collect_handle
+            .await
+            .expect("failed to stop collector thread");
+        writer_handle.await.expect("failed to stop writer thread");
+    }
+
+    async fn start_collect_worker(rx: Receiver<RecordMsg>, writer_tx: Sender<WriteMsg>) {
+        let mut records = Vec::new();
+        loop {
+            if let Ok(message) = rx.recv_async().await {
+                match message {
+                    RecordMsg::Left(record) => {
+                        records.push(record);
+                    }
+                    RecordMsg::Right(()) => {
+                        writer_tx
+                            .send(WriteMsg::Left(records))
+                            .expect("failed to send write req");
+                        writer_tx
+                            .send(WriteMsg::Right(()))
+                            .expect("failed to kill writer thread");
+                        return;
+                    }
+                }
+            }
+            if !records.is_empty() && !rx.is_empty() {
+                writer_tx
+                    .send(WriteMsg::Left(records))
+                    .expect("failed to send write req");
+                records = Vec::new();
+            }
+        }
+    }
+
+    async fn start_writer_worker(rx: Receiver<WriteMsg>, mut writer: Box<dyn TraceWriter + Send>) {
+        loop {
+            if let Ok(msg) = rx.recv_async().await {
+                match msg {
+                    WriteMsg::Left(records) => {
+                        writer
+                            .write_all(records)
+                            .expect("failed to write the log file");
+                    }
+                    WriteMsg::Right(()) => {
+                        writer.flush().expect("failed to flush content");
+                        return;
+                    }
+                }
+            }
+        }
+    }
+
+    fn finish(&self) {
+        self.tx
+            .send(Either::Right(()))
+            .expect("failed to finish collector");
+    }
+
+    fn tx(&self) -> Sender<RecordMsg> {
+        self.tx.clone()
+    }
+
+    #[cfg(test)]
+    fn rx(&self) -> Receiver<RecordMsg> {
+        self.rx.clone()
+    }
+}
+
+impl Drop for GlobalCollector {
+    fn drop(&mut self) {
+        self.finish();
+    }
+}
+
+/// `TraceSpan` traces hummock operations. It marks the beginning of an operation and
+/// the end when the span is dropped. So, please make sure the span live long enough.
+/// Underscore binding like `let _ = span` will drop the span immediately.
+#[must_use = "TraceSpan Lifetime is important"]
+#[derive(Clone)]
+pub struct TraceSpan {
+    tx: Option<Sender<RecordMsg>>,
```

Comment:
Any reason to make the field `Option`?

Reply:
Seems that it's for `none`. If so, we can remove the Option on the type, and return `None` on `TraceSpan::none`.